### PR TITLE
Convert FormComponent proptype

### PR DESCRIPTION
**Pull request by Ronolibert**
## Description

Make proptype for formComponent to be a function to avoid proptype error

## Setup

Add any steps or code to run in this section to help others prepare to run your code:

```sh
make server_run
make tsp_client_run
```

### CODE REVIEW

File: src/scenes/TransportationServiceProvider/FormButton.jsx [hunk 1]
```diff
@@ -40,7 +40,7 @@ export default class FormButton extends Component {
 }
 
 FormButton.propTypes = {
-  formComponent: PropTypes.element.isRequired,
+  FormComponent: PropTypes.func.isRequired,
```
Comment (macrael) [inline src/scenes/TransportationServiceProvider/FormButton.jsx]: With returning <FormComponent> does this still have to be func?

**Comment by Ronolibert** [comment]
If we wanted to keep the propType as an element then the solution to fix the warning would be something to the effect of:
```
<FormButton
  FormComponent={<PickupDateForm />}
  {...props}
/>

// FormButton.jsx

export class FormButton extends Component {
  enterFormView = () => {
    const { schema, onSubmit, FormComponent } = this.props;
    const formProps = {
      onCancel: this.setButtonState,
      schema: schema,
      onSubmit: onSubmit,
    };
    return React.cloneElement(FormComponent, [...formProps]);
  };
 render() { ...stuff }
}
```

File: src/scenes/TransportationServiceProvider/FormButton.jsx (hunk 1 above)
Comment (Ronolibert) [reply src/scenes/TransportationServiceProvider/FormButton.jsx]: Yeah it does since we're not really executing the function til enterFormView is called

**Comment by macrael** [comment]
Oh ok, I see that an uninstantiated element is not an element, which makes sense. This is better.

File: src/scenes/TransportationServiceProvider/FormButton.jsx (hunk 1 above)
Comment (macrael) [reply src/scenes/TransportationServiceProvider/FormButton.jsx]: 👍

**Pull request merged**